lib/routemaster/api_client.rb

```diff
@@ -98,7 +115,7 @@ def connection
         f.response :json, content_type: /\bjson/
         f.use Routemaster::Middleware::ResponseCaching, listener: @listener
         f.use Routemaster::Middleware::Metrics, client: @metrics_client, source_peer: @source_peer
-        f.adapter :net_http_persistent
+        f.adapter :typhoeus
```

Comment:
not really, because Typhoeus+Faraday supports a [form of parallelism](https://github.com/lostisland/faraday/wiki/Parallel-requests) but not concurrency as we need it (ie. futures).